### PR TITLE
feat:  Added  MatMul instance method to Tensor class of Paddle Frontend

### LINKED ISSUE #27759: __matmul__

### PR DESCRIPTION
**Pull request by Increshi**
# PR Description
Added matmul instance method to tensor class in paddle frontend.

## Related Issue

Closes #27759

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [ ] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Review by he11owthere — COMMENTED** [review]
Hey @Increshi, tests are failing for all the backends:
https://github.com/unifyai/ivy/actions/runs/7225118292/job/19688826244?pr=27763#step:6:61

Could you look into that? Thanks!

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 1]
```diff
@@ -904,6 +904,52 @@ def test_paddle__len__(
...
+    init_tree="paddle.to_tensor",
+    method_name="__matmul__",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        num_arrays=2,
+        shared_dtype=True,
+    ),
+    transpose_x=st.booleans(),
+    transpose_y=st.booleans(),
+    test_with_out=st.just(False),
```
Comment (NiteshK84) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:919]: this parameter test_with_out doesn't exist in the decorator, please remove this.

Comment (NiteshK84) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:919]: image

Comment (NiteshK84) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py]: But I don't see these functions implemented yet.
image
https://github.com/unifyai/ivy/issues/27759
The function you reserved by linking it to an open issue is still pending, so you can continue working on it until you close your PR.

Comment (NiteshK84) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py]: instance methods and dunder methods are different,
image

**Comment by NiteshK84** [comment]
Hi @Increshi,
please resolve the merge conflicts.

**Pull request closed without merging**